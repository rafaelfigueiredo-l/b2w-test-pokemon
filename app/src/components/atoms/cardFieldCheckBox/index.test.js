import CardFieldCheckBox from "./";

describe("<CardFieldCheckBox />", () => {
  const onChangeMock = jest.fn();

  const component = (
    <CardFieldCheckBox
      onChange={onChangeMock()}
      checked={0}
      placeholder="Something"
    />
  );

  const mounted = mount(component);
  const wrapper = shallow(component);

  it("should mount correctly CardFieldCheckBox component", () => {
    expect(mounted).toMatchSnapshot();
  });

  it("should have className cardFieldCheckBox", () => {
    expect(wrapper.find(".cardFieldCheckBox")).toHaveLength(1);
  });

  it("should call onChange function", () => {
    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toBeCalled();
  });

  it("should render input type checkbox", () => {
    expect(wrapper.find("[type='checkbox']")).toHaveLength(1);
  });

  it("should checked be false", () => {
    expect(wrapper.prop("checked")).toEqual(0);
  });

  it("should render correctly placeholder", () => {
    expect(wrapper.prop("placeholder")).toEqual("Something");
  });
});
