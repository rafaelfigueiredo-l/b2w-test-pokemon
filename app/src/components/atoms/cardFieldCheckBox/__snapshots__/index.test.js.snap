// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardFieldCheckBox /> should mount correctly CardFieldCheckBox component 1`] = `
<styled.input
  checked={0}
  className="cardFieldCheckBox"
  onChange={[Function]}
  placeholder="Something"
  type="checkbox"
>
  <input
    checked={0}
    className="sc-AxjAm dLgPKd cardFieldCheckBox"
    onChange={[Function]}
    placeholder="Something"
    type="checkbox"
  />
</styled.input>
`;
